app/response_processor.py

```diff
@@ -46,6 +48,30 @@ def process(self, message, tx_id=None):
 
         return
 
+    def _check_namespace_error(self, response):
+        """Takes a response from rrm receipt endpoint and examines the xml
+        to identify whether a RetryableError or QuarantinableError should
+        be raised"""
+        namespaces = {'error': 'http://ns.ons.gov.uk/namespaces/resources/error'}
+        tree = etree.fromstring(response.content)
+        element = tree.find('error:message', namespaces).text
+        elements = element.split('-')
+
+        if elements[0] == '1009':
+            stat_unit_id = elements[-1].split('statistical_unit_id: ')[-1].split()[0]
+            collection_exercise_sid = elements[-1].split(
+                'collection_exercise_sid: ')[-1].split()[0]
+
+            self.logger.error("Receipt rejected by endpoint",
+                              msg="No records were found on the man_ce_sample_map table",
+                              error=1009,
+                              stat_unit_id=stat_unit_id,
+                              collection_exercise_sid=collection_exercise_sid)
+            raise QuarantinableError
+        else:
+            self.logger.error("Bad response from endpoint")
```

Comment:
Better `Bad response from RRM`. `endpoint` is a bit nebulous

Reply:
I agree, but to do anything else would require further xml parsing.